roles/confluent.kafka_broker/defaults/main.yml

```diff
@@ -34,6 +34,15 @@ kafka_broker_sysctl_file: /etc/sysctl.conf
 
 kafka_broker_default_interal_replication_factor: "{{ [ groups['kafka_broker'] | length, 3 ] | min }}"
 
+kafka_broker_confluent_metrics:
+  enabled: true
+  servers: "{{groups['kafka_broker']}}"
+  listener: "{{kafka_broker_listeners['external']}}"
+kafka_broker_confluent_metrics_bootstrap_servers: "{% for host in kafka_broker_confluent_metrics.servers %}{% if loop.index > 1%},{% endif %}{{ host }}:{{kafka_broker_confluent_metrics.listener.port}}{% endfor %}"
+
+kafka_broker_metric_reporters: []
+kafka_broker_metric_reporters_final: "{{kafka_broker_metric_reporters + [kafka_broker_confluent_metrics_reporter_cls] if kafka_broker_confluent_metrics.enabled|bool else kafka_broker_metric_reporters}}"
```

Comment:
Im a little lost on this logic

so if kafka_broker_confluent_metrics.enabled|bool == false 
then kafka_broker_metric_reporters_final is an empty list  and there wont be any metrics properties

and if kafka_broker_confluent_metrics.enabled|bool == true
then  kafka_broker_metric_reporters_final has the default classes + any additional ones 
and there will be metrics properties

is there a scenario where you dont want kafka_broker_confluent_metrics_reporter_cls in  this list kafka_broker_metric_reporters_final

Reply:
will the broker error out if `metric.reporters` prop doesnt have any classes?

Reply:
`kafka_broker_confluent_metrics.enabled` is strictly for controlling only the confluent provided metrics reporters. `kafka_broker_metric_reporters` is for the end user to provide any of their own custom metrics reporters for integration with other systems outside of Control Center. So the logic for `kafka_broker_metric_reporters_final` is designed to always include whatever has been added to `kafka_broker_metric_reporters` and optionally add the Confluent Metrics Reporters based on the `enabled` flag.

As for the question about `metric.reporters`. The logic was done in this fashion as to address one of your earlier comments about not leaving it empty in the earlier PR code.